### PR TITLE
Add multiple possible displayed texts to button

### PR DESCRIPTION
**Pull request by fcpoloia**
it will convert the input text to list and iterate the list with each time `iterate_text_list()` is called, changing the displayed text to the next in list, it will loop around if reached the end of a list

### CODE REVIEW

**Comment by owalk** [comment]
This is a small thing but as general feedback, the commit title should be a small denotation of the scope of the commit, be on a single line, start with a capitalized letter, and have no punctuation at the end. I'll add this to a linter later, but it makes the commits much cleaner looking at a glance.

The commit bodies should be full sentences with proper punctuation, and further detail not in the commit title.

File: m1wengine/button.py [hunk 1]
```diff
@@ -136,6 +136,3 @@ def update(self):
         self.__text.render_font()
         self.image.blit(self.__text.image, (0, 0))
 
-    def update_and_iterate(self):
```
Comment (owalk) [inline m1wengine/button.py]: Code in a single PR should not get added and removed in two separate commits, or removed and added back in in two separate commits.  It adds significant time to the review process. If you need assistance with the git commands to clean your git history just let me know and I can walk you through it in discord or write out the steps and commands in this review comment

File: m1wengine/button.py [hunk 2]
```diff
@@ -65,33 +66,53 @@ def __init__(
...
 
         self.__fill_colors: dict[str, str] = {
             "normal": "#ffffff",
             "hover": "#666666",
             "pressed": "#333333",
         }
 
+    def iterate_text_list(self):
+        #Increase the index to change the displayed text and reset index if it's already on the last element
+                if self.Button_Texts_List[self.index] == self.Button_Texts_List[-1]:
```
Comment (owalk) [inline m1wengine/button.py:115]: indentation is wrong in this logic block. Most IDE's will let you highlight the logic you want to change the indentation on, and press shift tab to move it backwards, or tab to move it forwards.

The logic needs to be pulled back one level worth of indentation.

File: m1wengine/button.py [hunk 3]
```diff
@@ -65,33 +66,53 @@ def __init__(
...
+        self.__text: Text = Text(x, y, self.button_text)
 
         self.__fill_colors: dict[str, str] = {
             "normal": "#ffffff",
             "hover": "#666666",
             "pressed": "#333333",
         }
 
+    def iterate_text_list(self):
+        #Increase the index to change the displayed text and reset index if it's already on the last element
```
Comment (owalk) [inline m1wengine/button.py:114]: Change this comment to proper doc strings. 
```Suggestion
'''Increase the index to change the displayed text.

Reset the index if on the last element.
'''
```

File: m1wengine/button.py [hunk 4]
```diff
@@ -105,7 +126,12 @@ def update(self):
                 elif not self.__already_pressed:
                     self._on_click_function()
                     self.__already_pressed = True
+
```
Comment (owalk) [inline m1wengine/button.py:133]: formatting change in a non-formatting commit.

This is again a small thing and can be ignored for this PR. In general, it's nice to pull all formatting changes into a separate commit so they do no distract from code changes within the more technical commits.

File: m1wengine/button.py [hunk 5]
```diff
@@ -49,7 +49,7 @@ def __init__(
         y: int,
         width: int,
         height: int,
-        text: str,
+        text,
```
Comment (owalk) [inline m1wengine/button.py:52]: I would squash this entire commit onto the previous commit as it is within the same scope of the initial commit. Doing it this way just adds time to the PR review process.

File: m1wengine/button.py (hunk 1 above)
Comment (owalk) [reply m1wengine/button.py]: this commit can be squashed into the initial commit. alongside the second commit. At the end, there only needs to be 1 commit in this PR, unless you decide to pull out the whitespace change into its own formatting commit.

**Review by owalk — CHANGES_REQUESTED** [review]
I haven't reviewed the logic or organization of the code yet. The first focus is on the commit scope issues and formatting issues.

**Comment by owalk** [comment]
I've enabled the GitHub actions so that future pushes will have lint checks run against them.

File: m1wengine/button.py (hunk 1 above)
Comment (owalk) [reply m1wengine/button.py]: To squash commits, I recommend doing the following:

Git rebase -i HEAD~5
# When your text editor opens, mark the commits to squash with the letter 's'
# save and close the text editor
# edit the commit title and body to be scoped to the new resulting commit
# overwrite the changes on the remote branch with a force push
git push origin 'branch-name' --force

Comment (fcpoloia) [reply m1wengine/button.py]: I know how to squash commits, this just something weird happened, I must've not save the config correctly, I don't know what else might've went wrong

File: m1wengine/button.py [hunk 6]
```diff
@@ -65,33 +66,57 @@ def __init__(
             The width of the button
         height: int
             The height of the button
-        button_text
-            The string to display on the button
+        text: str
+            The string or list to be displayed on the button
+        splitter: str
+            a char that would split the string into separate elements in list, leave empty if you only want 1 element
```
Comment (owalk) [inline m1wengine/button.py:72]: this line is to long. the linter is set to a max line length of 88 characters, and it will complain when you push the code again.

To test this locally, run 
flake8 --max-line-length=88 .

image

File: m1wengine/button.py [hunk 7]
```diff
@@ -65,33 +66,57 @@ def __init__(
...
+        splitter: str
+            a char that would split the string into separate elements in list, leave empty if you only want 1 element 
         on_click_function: callable
             The method to run when the button is clicked
         one_press: bool
             Should the button click once or call repeatedly when held
         """
+
         super().__init__()
+        #Splits text into a list
```
Comment (owalk) [inline m1wengine/button.py:80]: space after the hashtag on comments

File: m1wengine/button.py [hunk 8]
```diff
@@ -65,33 +66,57 @@ def __init__(
...
+        self.button_text = ''
+        if type(text) == str:
+            if splitter != '':
+                self.Button_Texts_List = text.split(splitter)
+            else:
+                self.Button_Texts_List = [text]
+        elif type(text) == list:
+            self.Button_Texts_List = text
+        self.button_text = self.Button_Texts_List[self.index]
+        #Dimensions
```
Comment (owalk) [inline m1wengine/button.py:92]: space after hashtag on comments

File: m1wengine/button.py [hunk 9]
```diff
@@ -65,33 +66,57 @@ def __init__(
...
         self.__x: int = x
         self.__y: int = y
         self.__width: int = width
         self.__height: int = height
+        #Clicking functionaliti
```
Comment (owalk) [inline m1wengine/button.py:97]: missing space

File: m1wengine/button.py [hunk 10]
```diff
@@ -65,33 +66,57 @@ def __init__(
...
+        self.__text: Text = Text(x, y, self.button_text)
 
         self.__fill_colors: dict[str, str] = {
             "normal": "#ffffff",
             "hover": "#666666",
             "pressed": "#333333",
         }
 
+    def iterate_text_list(self):
+        'Increase the index to change the displayed text and reset index if its already on the last element'
```
Comment (owalk) [inline m1wengine/button.py:114]: to keep the docstrings consistent they should open and close with 3 of quotation mark characters

The line length is probably also too long, check with the linter.

```Suggestion
        """Increase the index to change the displayed text and reset index if its already on the last element'"""
```

File: m1wengine/button.py [hunk 11]
```diff
@@ -109,3 +134,4 @@ def update(self):
                 self.__already_pressed = False
         self.__text.render_font()
         self.image.blit(self.__text.image, (0, 0))
+
```
Comment (owalk) [inline m1wengine/button.py:137]: the linter is set to disallow new lines at the end of the file

**Review by owalk — CHANGES_REQUESTED** [review]
Here's some of the old readme which has not been pulled into the current repository yet.

You might want to start using the linter script to check your code, it's a lot easier

image

File: m1wengine/button.py (hunk 11 above)
Comment (owalk) [reply m1wengine/button.py:137]: this one was actually about white spaces, not the new line.

**Review by owalk — CHANGES_REQUESTED** [review]
Try to get all the changes on a single commit again since it is all in the scope of that first initial commit, and try to get the lint check to pass. You can press "details" to view what is failing. You can also run the "./linter.sh" script to run the checks locally. I have it set up to run that linter check every time I hit save, and find that very helpful.

image

File: m1wengine/button.py (hunk 11 above)
Comment (owalk) [reply m1wengine/button.py:138]: image

File: m1wengine/button.py [hunk 12]
```diff
@@ -65,33 +66,58 @@ def __init__(
...
+        self.__text: Text = Text(x, y, self.button_text)
 
         self.__fill_colors: dict[str, str] = {
             "normal": "#ffffff",
             "hover": "#666666",
             "pressed": "#333333",
         }
 
+    def iterate_text_list(self):
+        """Increase the index to change the displayed text and reset index if its already on the last element"""
```
Comment (owalk) [inline m1wengine/button.py:115]: image

```Suggestion
        """Iterate through the display button text.
        
        Increase the index to change the displayed text and
        reset index if it's already on the last element
        """
```

File: m1wengine/button.py [hunk 13]
```diff
@@ -103,6 +103,7 @@ def __init__(
             self.__x, self.__y, self.__width, self.__height
         )
         self.__text: Text = Text(x, y, self.button_text)
+        self.__text: Text = Text(x, y, self.button_text)
```
Comment (owalk) [inline m1wengine/button.py:106]: It would be nice to not have any merge commits. This can be squashed into a single commit for the PR

**Review by owalk — CHANGES_REQUESTED** [review]

File: m1wengine/button.py (hunk 11 above)
Comment (owalk) [reply m1wengine/button.py:138]: run the black formatter and this will be fixed automatically. you may need to install it through the requirments.txt if you haven't yet

File: m1wengine/button.py (hunk 13 above)
Comment (fcpoloia) [reply m1wengine/button.py:106]: I tried squashing it but it always ends up with either merge error or creates this commit anyway

Comment (owalk) [reply m1wengine/button.py:106]: I need a bit more information to help with this. What are the exact steps to squash that come back with a merge error?

File: m1wengine/button.py [hunk 14]
```diff
@@ -65,33 +63,61 @@ def __init__(
             The width of the button
         height: int
             The height of the button
-        button_text
-            The string to display on the button
+        text: str
+            The string or list to be displayed on the button
+        splitter: str
+            a char that would split the string into separate elements
```
Comment (Sean-Nishi) [inline m1wengine/button.py:69]: Docstrings should start with a capitalized letter

File: m1wengine/button.py [hunk 15]
```diff
@@ -65,33 +63,61 @@ def __init__(
...
             The method to run when the button is clicked
         one_press: bool
             Should the button click once or call repeatedly when held
         """
+
         super().__init__()
+        # Splits text into a list
+        self.index = 0
+        self.Button_Texts_List = []
+        self.button_text = ''
```
Comment (Sean-Nishi) [inline m1wengine/button.py:80]: Variables need type hints (even if they are obvious)

File: m1wengine/button.py [hunk 16]
```diff
@@ -65,33 +63,61 @@ def __init__(
...
+        self.button_text = self.Button_Texts_List[self.index]
+        # Dimensions
         self.__x: int = x
         self.__y: int = y
         self.__width: int = width
         self.__height: int = height
+        # Clicking functionaliti
         self._on_click_function: callable = on_click_function
         self.__on_press: bool = one_press
         self.__already_pressed: bool = False
```
Comment (Sean-Nishi) [inline m1wengine/button.py:97]: These look good except for the variable mangling. We are trying to follow Google's style guide as closely as possible. Please remove one of the underscores.
You can read more about the styling of "private" variables in python at: https://google.github.io/styleguide/pyguide.html#3162-naming-conventions
tldr: Use only single prepending _ on private variables to represent they are "private" (no variable in python is actually considered private, this is a style choice to help the reader/coder). Then if other classes want to access these private variables add in the @property and @property.setter methods. There should be a few examples already in our codebase.

File: m1wengine/button.py [hunk 17]
```diff
@@ -65,33 +63,61 @@ def __init__(
...
         self.image: pygame.Surface = pygame.Surface((self.__width, self.__height))
         self.rect: pygame.Rect = pygame.Rect(
             self.__x, self.__y, self.__width, self.__height
         )
-        self.__text: Text = Text(x, y, button_text)
+        self.__text: Text = Text(x, y, self.button_text)
+        self.__text: Text = Text(x, y, self.button_text)
```
Comment (Sean-Nishi) [inline m1wengine/button.py:103]: Same comment as above. Only have one prepending underscore

**Review by Sean-Nishi — CHANGES_REQUESTED** [review]
linter
You are so close! Just a few stylistic changes to make sure we are adhering to some sort of standard.
Also, for the failed lint check, once you make the changes I request if you run the command "./linter.sh" into your development environment then black will tell you what is wrong with the styling in this file. Make those additional changes and I will re-review. Then it should be good to go.

File: m1wengine/button.py (hunk 15 above)
Comment (owalk) [inline m1wengine/button.py:80]:
```Suggestion
        # Splits text into a list
        self.index: int = 0
        self.Button_Texts_List: list = []
        self.button_text: str = ''
```

File: m1wengine/button.py [hunk 18]
```diff
@@ -65,33 +63,61 @@ def __init__(
...
+                self.Button_Texts_List = [text]
+        elif type(text) == list:
+            self.Button_Texts_List = text
+        self.button_text = self.Button_Texts_List[self.index]
+        # Dimensions
         self.__x: int = x
         self.__y: int = y
         self.__width: int = width
         self.__height: int = height
+        # Clicking functionaliti
```
Comment (owalk) [inline m1wengine/button.py:94]:
```Suggestion
        # Clicking functionality
```

File: m1wengine/button.py (hunk 16 above)
Comment (owalk) [reply m1wengine/button.py:97]: plus one, I honestly forgot about this ^ 

As a Python newbie not having the concept of private variables is so strange. Pretty much any doc online I checked agrees with the Google style of not mangling privates -_-

**Review by owalk — CHANGES_REQUESTED** [review]

File: m1wengine/button.py [hunk 19]
```diff
@@ -1,6 +1,3 @@
-"""This module contains the Button class."""
```
Comment (owalk) [inline m1wengine/button.py:1]: Removing this docstring is causing the linter to fail here

File: m1wengine/button.py [hunk 20]
```diff
@@ -65,33 +63,61 @@ def __init__(
...
+        text: str
+            The string or list to be displayed on the button
+        splitter: str
+            a char that would split the string into separate elements
         on_click_function: callable
             The method to run when the button is clicked
         one_press: bool
             Should the button click once or call repeatedly when held
         """
+
```
Comment (owalk) [inline m1wengine/button.py:75]: this blank line causes the linter to fail

File: m1wengine/button.py [hunk 21]
```diff
@@ -65,33 +63,61 @@ def __init__(
...
 
         self.__fill_colors: dict[str, str] = {
             "normal": "#ffffff",
             "hover": "#666666",
             "pressed": "#333333",
         }
 
+    def iterate_text_list(self):
+        """Iterate through the display button text.
+        Increase the index to change the displayed text and
```
Comment (owalk) [inline m1wengine/button.py:113]: the lack of newline here causes the linter to fail
